src/data/finset/basic.lean

```diff
@@ -11,19 +11,97 @@ import tactic.nth_rewrite
 /-!
 # Finite sets
 
-mathlib has several different models for finite sets,
-and it can be confusing when you're first getting used to them!
+What are `finset`s in mathlib?
 
-This file builds the basic theory of `finset α`,
-modelled as a `multiset α` without duplicates.
+Terms of type `finset α` are subsets of `α` which are finite.
+In Lean, `finset α` is implemented to be 2 pieces of data:
 
-It's "constructive" in the since that there is an underlying list of elements,
-although this is wrapped in a quotient by permutations,
-so anytime you actually use this list you're obligated to show you didn't depend on the ordering.
+  1. `val` is a `multiset α` of elements
+  2. `no_dup` is a proof that `val` has no duplicates.
+
+Finsets in Lean are constructive since you must exhibit
+a list of their underlying elements to use them.
```

Comment:
I might change this sentence so that it doesn't seem like *you* need to think about these lists.
```text
Finsets in Lean are constructive in that they have an underlying list that enumerates
their elements.
```